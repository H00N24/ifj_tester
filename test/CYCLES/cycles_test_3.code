// Rozsirenie CYCLES

class Main
{
	static void run()
	{
		ifj16.print("---- START ----" + "\n");

		Tri.test();

		ifj16.print("---- KONIEC ----" + "\n");
	}
}

class Tri
{
	static void test()
	{
		double v;
		for (int i = 5; i < 10; i = i + 1)
		{
			Dva.setA();
			ifj16.print("A set: " + Dva.a + "\n");
			
			v = Dva.rekurz();
			ifj16.print(i + ". Vysledok: " + v + "\n");
		}

		ifj16.print("Idem do Main.run()" + "\n");

		return;
	}
}

class Dva
{
	static int a = 5;

	static void setA()
	{
		a = 5;
	}

	static int rekurz()
	{
		int c;
		if (a > 0)
		{
			ifj16.print("R: " + a + "\n");
			a = Dva.a - 1;
			
			c = rekurz();
			return c;
		}

		return 128;
	}
}
