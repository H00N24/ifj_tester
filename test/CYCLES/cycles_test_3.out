---- START ----
A set: 5
R: 5
R: 4
R: 3
R: 2
R: 1
5. Vysledok: 128
A set: 5
R: 5
R: 4
R: 3
R: 2
R: 1
6. Vysledok: 128
A set: 5
R: 5
R: 4
R: 3
R: 2
R: 1
7. Vysledok: 128
A set: 5
R: 5
R: 4
R: 3
R: 2
R: 1
8. Vysledok: 128
A set: 5
R: 5
R: 4
R: 3
R: 2
R: 1
9. Vysledok: 128
Idem do Main.run()
---- KONIEC ----
